From 841fc1e3a748ce8b2b46df08198cc8673c27eb18 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 3 Mar 2015 18:20:44 +0100
Subject: [PATCH] base/init: don't exit the main loop with waiting jobs

unfinished jobs in the queue can cause unforseen side effects
---
 modules.d/99base/init | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/99base/init b/modules.d/99base/init
index 3d91e76..7b89083 100755
--- a/modules.d/99base/init
+++ b/modules.d/99base/init
@@ -182,7 +182,6 @@ while :; do
     for job in /initqueue/*.sh; do
 	[ -e "$job" ] || break
         job=$job . $job
-        check_finished && break 2
     done
 
     $UDEV_QUEUE_EMPTY >/dev/null 2>&1 || continue
@@ -194,11 +193,12 @@ while :; do
     for job in /initqueue-settled/*.sh; do
         [ -e "$job" ] || break
         job=$job . $job
-        check_finished && break 2
     done
 
     $UDEV_QUEUE_EMPTY >/dev/null 2>&1 || continue
 
+    check_finished && break
+
     # no more udev jobs and queues empty.
     sleep 0.5
 
